From ff67878dc49a1f5e2489d43c175d99f61045cb7c Mon Sep 17 00:00:00 2001
From: Rob Evers <revers@redhat.com>
Date: Fri, 27 Jun 2014 19:10:00 -0400
Subject: [scsi] be2iscsi: fix memory leak in error path

Message-id: <1403896200-3787-1-git-send-email-revers@redhat.com>
Patchwork-id: 85081
O-Subject: [RHEL7.1 e-stor PATCH v3 08/10] scsi: be2iscsi: remove potential junk pointer free
Bugzilla: 1088576
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Maurizio Lombardi <mlombard@redhat.com>
RH-Acked-by: Tomas Henzl <thenzl@redhat.com>

From: John Soni Jose <sony.john-n@emulex.com>

commit 0e7c60c [SCSI] be2iscsi: fix memory leak in error path
fixed an potential junk pointer free if  mgmt_get_if_info() returned an error

fix it on one more place

http://marc.info/?l=linux-scsi&m=140205737008785&w=2

Signed-off-by: Tomas Henzl <thenzl@redhat.com>
Reviewed-by: Mike Christie <michaelc@cs.wisc.edu>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/be2iscsi/be_mgmt.c b/drivers/scsi/be2iscsi/be_mgmt.c
index 6045aa7..07934b0 100644
--- a/drivers/scsi/be2iscsi/be_mgmt.c
+++ b/drivers/scsi/be2iscsi/be_mgmt.c
@@ -1008,10 +1008,8 @@ int mgmt_set_ip(struct beiscsi_hba *phba,
   BE2_IPV6 : BE2_IPV4 ;
 
  rc = mgmt_get_if_info(phba, ip_type, &if_info);
- if (rc) {
-  kfree(if_info);
+ if (rc)
   return rc;
- }
 
  if (boot_proto == ISCSI_BOOTPROTO_DHCP) {
   if (if_info->dhcp_state) {
-- 
1.7.1